rholang/src/main/scala/coop/rchain/rholang/interpreter/accounting/package.scala

```diff
@@ -1,23 +1,56 @@
 package coop.rchain.rholang.interpreter
 
-import cats.Monad
+import cats._
+import cats.data._
+import cats.effect.Sync
+import cats.effect.concurrent.Ref
 import cats.implicits._
-import cats.mtl.MonadState
+import cats.mtl._
 
 import coop.rchain.rholang.interpreter.errors.OutOfPhlogistonsError
 
 package object accounting extends Costs {
 
-  type _cost[F[_]] = MonadState[F, Cost]
-  def _cost[F[_]](implicit ev: _cost[F]): _cost[F] = ev
+  type _cost[F[_]]       = MonadState[F, Cost]
+  type _costWriter[F[_]] = FunctorTell[F, Chain[Cost]]
+
+  def _cost[F[_]](implicit ev: _cost[F]): _cost[F]                   = ev
+  def _costWriter[F[_]](implicit ev: _costWriter[F]): _costWriter[F] = ev
 
   def charge[F[_]: Monad](
       amount: Cost
-  )(implicit cost: _cost[F], error: _error[F]): F[Unit] =
+  )(implicit cost: _cost[F], error: _error[F], writer: _costWriter[F]): F[Unit] =
     for {
       _ <- error.ensure(cost.get)(OutOfPhlogistonsError)(_.value >= 0)
+      _ <- writer.tell(Chain.one(amount))
       _ <- cost.modify(_ - amount)
       _ <- error.ensure(cost.get)(OutOfPhlogistonsError)(_.value >= 0)
     } yield ()
 
+  def costWriter[M[_]: Sync](): FunctorListen[M, Chain[Cost]] =
```

Comment:
I think 'FunctorTell' should suffice - we should only ever need the cost value at the end of the world.